docs/standard/microservices-architecture/multi-container-microservice-net-applications/implementing-api-gateways-with-ocelot.md

```diff
@@ -0,0 +1,579 @@
+
+```csharp
+FROM microsoft/aspnetcore:2.0.5 AS base
+WORKDIR /app
+EXPOSE 80
+```
+The port 80 shown in the code is internal within the Docker host, so it cannot be reached by the client apps. 
+The client apps can access only to the external ports (if any) published when deploying with docker-compose.
+
+Those external ports should not be published when deploying into a production environment. This is precisely why you want to use the API Gateway, to avoid the direct communication between the client apps and the microservices.
+
+However, when developing, you want to access the microservice/container directly and run it through Swagger. That’s why in eShopOnContainers the external ports are still specified even when they won’t be used by the API Gateway or the client apps.
+
+Here’s an example of the docker-compose.override.yml file for the Catalog microservice.
+
+```csharp
+catalog.api:
+  environment:
+    - ASPNETCORE_ENVIRONMENT=Development
+    - ASPNETCORE_URLS=http://0.0.0.0:80
+    - ConnectionString=YOUR_VALUE
+    - ... Other Environment Variables
+  ports:
+    - "5101:80"   # Important: In a production environment you should remove the external port (5101) kept here for microservice debugging purposes. 
+                  # The API Gateway redirects and access through the internal port (80).
+```
+
+You can see how in the docker-compose.override.yml configuration the internal port for the Catalog container is port 80, but the port for external access is 5101. But this port shouldn’t be used by the application when using an API Gateway, only to debug run and test just the Catalog microservice.
+
+Normally, you won’t be deploying with docker-compose into a production environment because the right production deployment environment for microservices is an orchestrator like Kubernetes or Service Fabric and when deploying to those environments you will use different configuration files where you won’t publish directly any external port for the microservices but you will always use the reverse proxy from the API Gateway.
+
+Let’s run the catalog microservice in your local Docker host either by running the full eShopOnContainers solution from Visual Studio (it’ll run all the services in the docker-compose files) or just starting the Catalog microservice with the following docker-compose command in CMD or PowerShell positioned at the folder where the docker-compose.yml and docker-compose.override.yml are placed.
+
+```csharp
+docker-compose run --service-ports catalog.api
+```
+
+This command will only run the catalog.api service container plus its dependencies which are specified in the docker-compose.yml.
+In this case, the dependencies are the SQL Server container and RabbitMQ container.
+
+Then you could directly access the Catalog microservice and see its methods through the Swagger UI accessing directly through that “external” port, in this case http://localhost:5101 
```

Comment:
Then you could -> Then, you can

in this case sentence should be a separate one and address should be inside backticks because we don't use bare URLs

Reply:
👍